### PR TITLE
feat(services): add YAML service

### PR DESCRIPTION
**Pull request by aafre**
Create yamlService.ts with three core functions for YAML export/import:
- processSectionsForExport: Cleans icon paths and removes temporary fields
- exportResumeAsYAML: Exports resume data as YAML with embedded icons
- importResumeFromYAML: Imports and validates YAML with icon restoration

Features:
- Removes iconFile and iconBase64 fields during export
- Cleans "/icons/" prefix from icon paths (e.g., "/icons/logo.png" → "logo.png")
- Embeds icons as base64 in __icons__ section for portability
- Imports icons back into iconRegistry during load
- Migrates legacy sections (auto-adds type property)
- Handles icon-list, experience, and education sections

Test coverage:
- 32 comprehensive tests covering all functions
- 98.14% statement coverage (exceeds 90% requirement)
- 86.66% branch coverage
- 100% function coverage
- Tests icon embedding/extraction, edge cases, and error handling

All 469 tests passing across entire test suite.

Implements TASK-006 and TASK-007 from Editor.tsx refactoring plan.

### CODE REVIEW

File: resume-builder-ui/src/services/yamlService.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        return {
+          certification: cleanItem.certification || "",
+          issuer: cleanItem.issuer || "",
+          date: cleanItem.date || "",
+          icon: cleanItem.icon
+            ? cleanItem.icon.startsWith("/icons/")
+              ? cleanItem.icon.replace("/icons/", "")
+              : cleanItem.icon
+            : null,
+        };
```
[the remark this answers is not in the document]
Comment (aafre) [reply resume-builder-ui/src/services/yamlService.ts]: @gemini-code-assist - made it generic and updated the test.

File: resume-builder-ui/src/services/yamlService.ts [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+                : null,
+            };
+          })
+        : section.content;
+
+      return {
+        ...section,
+        content: updatedContent,
+      };
+    }
```
[the remark this answers is not in the document]
Comment (aafre) [reply resume-builder-ui/src/services/yamlService.ts]: @gemini-code-assist - done, I've consolidated the logic. Check now.
